### ISSUE TITLE
Scan agent incorrectly marking IPs as discovered that are unused

### ISSUE THREAD

**Issue by 9strands** [phpipam/phpipam-agent#74]
Versions:
PHPIPAM host: PHP IPAMversion 1.5 with PHP 7.3.20
PHPIPAM-agent host https://github.com/phpipam/phpipam-agent master branch as of 2021-08-29 with PHP 7.4.22

We're trying to scan a /22.  When I run the scan agent, it manually detects which IPs are up and not up according to the debug output when I run the scan manually.  However, in the PHPIPAM dashboard, all IP's in the subnet are marked as discovered.

For example - running php index.php update prints the following details out:

ICMP Host Unreachable from 172.23.241.13 for ICMP Echo sent to 172.23.240.3
ICMP Host Unreachable from 172.23.241.13 for ICMP Echo sent to 172.23.240.9
ICMP Host Unreachable from 172.23.241.13 for ICMP Echo sent to 172.23.240.8
ICMP Host Unreachable from 172.23.241.13 for ICMP Echo sent to 172.23.240.7
ICMP Host Unreachable from 172.23.241.13 for ICMP Echo sent to 172.23.240.6
ICMP Host Unreachable from 172.23.241.13 for ICMP Echo sent to 172.23.240.5
ICMP Host Unreachable from 172.23.241.13 for ICMP Echo sent to 172.23.240.4
...
172.23.240.1   : xmt/rcv/%loss = 1/1/0%, min/avg/max = 0.706/0.706/0.706
172.23.240.2   : xmt/rcv/%loss = 1/1/0%, min/avg/max = 0.739/0.739/0.739
172.23.240.3   : xmt/rcv/%loss = 1/0/100%
172.23.240.4   : xmt/rcv/%loss = 1/0/100%
172.23.240.5   : xmt/rcv/%loss = 1/0/100%
172.23.240.6   : xmt/rcv/%loss = 1/0/100%
172.23.240.7   : xmt/rcv/%loss = 1/0/100%
172.23.240.8   : xmt/rcv/%loss = 1/0/100%
...

However, in the subnet, these IPs all show up as online after the scan agent connects to the database and provides an update.

![image](https://user-images.githubusercontent.com/12106718/131639129-41a9d87a-89aa-44dc-aec9-753190c94f33.png)

**Comment by superusergoose** [comment]
I'm experiencing the same issue, I have not yet found a solution

**Comment by jsblock78** [comment]
I ran into this issue when method was fping, however with pear I got expected results.

**Comment by WagoL** [comment]
> I ran into this issue when method was fping, however with pear I got expected results.

How is your config with pear?

**Comment by jackychan114** [comment]
I have the same problem, which has not been solved yet.

**Comment by DPxCruelty** [comment]
My system environment is Rocky linux 9.3+PHP8.0, after debugging, I found that when **class.Scon.php** is scanned using fping, no matter whether the IP is reachable or not, there is a return value in the output, and the logic doesn't filter the timeout entries, I added an if function to deal with it, and it seems to be running normally at the moment!

**Comment by DPxCruelty** [comment]
My system environment is Rocky linux 9.3+PHP8.0, after debugging, I found that when **class.Scon.php** is scanned using fping, no matter whether the IP is reachable or not, there is a return value in the output, and the logic doesn't filter the timeout entries, I added an if function to deal with it, and it seems to be running normally at the moment!
In class.Scon.php on line 449
`if(sizeof($output)>0) {
                foreach($output as $line) {
                        $timeout = "timed out";
                        if(strpos($line, $timeout) !== false){
                        }else{
                                $tmp = explode(" ",$line);
                                $out[] = $tmp[0];
                        }
                }
            }`
